packages/expo-modules-core/android/src/androidTest/java/expo/modules/core/logging/PersistentFileLogTest.kt

```diff
@@ -0,0 +1,73 @@
+package expo.modules.core.logging
+
+import androidx.test.platform.app.InstrumentationRegistry
+import junit.framework.TestCase
+import org.junit.Assert
+import org.junit.Test
+
+class PersistentFileLogTest : TestCase() {
+  @Test
+  fun testPersistentLog() {
+    val instrumentationContext = InstrumentationRegistry.getInstrumentation().context
+    val persistentLog = PersistentFileLog("dev.expo.modules.core.logging.test", instrumentationContext)
+
+    asyncMethodRunning = true
+    persistentLog.clearEntries {
+      Assert.assertNull(it)
```

Comment:
we should also add a check to ensure that the callback is called on the same thread that it is called from

Reply:
Callbacks happening in Kotlin coroutines are not bound to a specific thread.